lib/resources/grub_conf.rb

```diff
@@ -71,35 +73,73 @@ def to_s
   ######################################################################
 
   def grub2_parse_kernel_lines(content, conf)
-    # Find all "menuentry" lines and then parse them into arrays
-    menu_entry = 0
+    menu_entries = extract_menu_entries(content)
+
+    if @kernel == 'default'
+      default_menu_entry(menu_entries, conf['GRUB_DEFAULT'])
+    else
+      menu_entries.select { |entry| entry['name'] == @kernel }[0]
```

Comment:
If we're always grabbing the first, we could change the `select` to a `find` which always returns the first element where the block evals `true`

Reply:
Neat! I didn't know that, will update.